src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -14,30 +14,165 @@
+        return Optional.of(lecture);
+      }
+    }
+    return Optional.empty();
+  }
+
+  @VisibleForTesting
+  /** Creates and returns {@code lectureEntity} using parameters found in {@code request}. */
+  // TODO: Send errors as a response if fields are empty.
+  protected Entity createLectureEntity(HttpServletRequest request) {
+    Optional<String> optionalLectureName = getParameter(request, NAME_INPUT);
+    String lectureName = optionalLectureName.isPresent() ? optionalLectureName.get() : "";
+    Optional<String> optionalVideoUrl = getParameter(request, LINK_INPUT);
+    String videoUrl = optionalVideoUrl.isPresent() ? optionalVideoUrl.get() : "";
+    Optional<String> optionalVideoId = getVideoId(videoUrl);
+    String videoId = optionalVideoId.isPresent() ? optionalVideoId.get() : "";
+
+    Entity lectureEntity = new Entity(Lecture.ENTITY_KIND);
+    lectureEntity.setProperty(Lecture.PROP_NAME, lectureName);
+    lectureEntity.setProperty(Lecture.PROP_VIDEO_URL, videoUrl);
+    lectureEntity.setProperty(Lecture.PROP_VIDEO_ID, videoId);
+    return lectureEntity;
+  }
+
+  /** Returns lectures stored in the database. */
+  private List<Lecture> getLectures() {
+    Query query = new Query(Lecture.ENTITY_KIND);
+    PreparedQuery results = datastore.prepare(query);
+    List<Lecture> lectures = new ArrayList<>();
+    for (Entity lecture : results.asIterable()) {
+      Lecture newLecture = Lecture.fromLectureEntity(lecture);
+      lectures.add(newLecture);
+    }
+    return lectures;
+  }
+
+  /**
+   * Returns the value of {@code name} from the {@code request} form.
+   * If the {@code name} cannot be found, return {@code Optional.empty()}.
+   */
+  private Optional<String> getParameter(HttpServletRequest request, String name) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return Optional.empty();
+    }
+    return Optional.of(value);
+  }
+
+  @VisibleForTesting
+  /** Returns YouTube video ID for a given {@code videoUrl}. */
+  protected Optional<String> getVideoId(String videoUrl) {
+    Matcher matcher = videoUrlGeneratedPattern.matcher(videoUrl);
+    if (matcher.find()) {
+      return Optional.of(matcher.group());
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Returns URL redirecting to lecture view page with parameters {@code lectureId}
+   * and {@code videoId} found in {@code lectureEntity}.
+   */
+  private Optional<String> buildRedirectUrl(Entity lectureEntity) {
+    String lectureId = String.valueOf(lectureEntity.getKey().getId());
+    String videoId = (String) lectureEntity.getProperty(Lecture.PROP_VIDEO_ID);
+
+    try {
+      URIBuilder urlBuilder = new URIBuilder(REDIRECT_URL)
+                                  .addParameter(Lecture.PROP_ID, lectureId)
+                                  .addParameter(Lecture.PROP_VIDEO_ID, videoId);
+      return Optional.of(urlBuilder.build().toString());
+    } catch (URISyntaxException urlBuilderError) {
```

Comment:
Should you print this error or something? Is it important to be made aware of this?

Reply:
Yes, handling errors in a different pr

Added todo